modules/lang-mustache/src/test/java/org/elasticsearch/script/mustache/SearchTemplateRequestXContentTests.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script.mustache;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParseException;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.script.ScriptType;
+import org.elasticsearch.test.AbstractXContentTestCase;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertToXContentEquivalent;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.nullValue;
+
+public class SearchTemplateRequestXContentTests extends AbstractXContentTestCase<SearchTemplateRequest> {
+
+    @Override
+    public SearchTemplateRequest createTestInstance() {
+        return SearchTemplateRequestTests.createRandomRequest();
+    }
+
+    @Override
+    protected SearchTemplateRequest doParseInstance(XContentParser parser) throws IOException {
+        return SearchTemplateRequest.fromXContent(parser);
+    }
+
+    /**
+     * Note that for xContent parsing, we don't check two parts of the request:
+     * - The 'simulate' option, since this parameter is not included in the
+     *   request's xContent (it's instead used to determine the request endpoint).
+     * - The random SearchRequest, since this component only affects the request
+     *   parameters and also isn't captured in the request's xContent.
+     */
+    @Override
+    protected void assertEqualInstances(SearchTemplateRequest expectedInstance, SearchTemplateRequest newInstance) {
+        expectedInstance.setSimulate(false);
+        newInstance.setSimulate(false);
+
+        expectedInstance.setRequest(null);
+        newInstance.setRequest(null);
+
+        super.assertEqualInstances(expectedInstance, newInstance);
```

Comment:
nit: maybe instead of mutating the two instances we should have the proper comparisons? I guess it is a subset of what equals and hashcode do?

Reply:
Thanks, I updated this and it looks clearer... we don't need the full equals and hashCode comparison because we're not supporting `testEqualsAndHashcode`.